FirebaseSwiftUI/FirebaseAuthSwiftUI/Sources/FirebaseAuthSwiftUI/FirebaseAuthSwiftUI.swift

```diff
@@ -0,0 +1,102 @@
+// The Swift Programming Language
+// https://docs.swift.org/swift-book
+import SwiftUI
+import FirebaseAuth
+import Combine
+
+
+protocol AuthListenerProtocol {
+    func onError(_ error: Error)
+    func onBeforeSignIn()
+    func onSignedIn(_ user: User)
+    func onCanceled()
+    // TODO - add when I get to this point
+     func onCredentialReceived(_ credential: AuthCredential)
+     func onCredentialLinked(_ credential: AuthCredential)
+    // func onMFARequired(_ resolver: MultiFactorResolver)
+}
```

Comment:
Since we're targeting this toward SwiftUI users, what do you think of adding a `@StateObject` that tracks the auth state so consumers can use it directly instead of having to set up their own wrapper? This could be the auth provider itself or a separate object that's managed by the auth provider, or a default implementation of the `AuthListenerProtocol`.

Reply:
Hey @morganchen12 - sorry for the lates response, I was on holiday last week, and this week I wanted something more concrete to show you so we can agree on top level API design/flow. I still need to refactor a couple of things, but I'm hoping by the end of the day tomorrow I can provide how it works and see if you think it is reasonable. Also - agree on the point of having some form of a state object the user can consume for auth events 👍